### DISCUSSION TITLE
Certain passwords stop working after client reboot

### DISCUSSION THREAD

**Discussion by felix-redlich** [rustdesk/rustdesk#9232] [General]
### Bug Description

When entering certain passwords as a Permanent Password, they work as intended until the system is rebooted.
After a reboot of the client, the client no longer accepts the password.

**Retained passwords**
The client accepts setting this password as a Permanent Password and allows starting sessions with it before a reboot:

`Ab1234-5678`

<img width="413" src="https://github.com/user-attachments/assets/a5a3c73f-8fc9-4ab7-bbd4-edd1bf251025">

After a reboot of the client, bullet characters are shown matching the length of the password:

<img width="410" src="https://github.com/user-attachments/assets/e71b133c-4b14-4616-a75b-62fe3af607b7">

This password is retained after a reboot and allows starting sessions normally.

**Unretained passwords**
The client also accepts setting this password as a Permanent Password and allows starting sessions with it before a reboot:

`D5K2Ulg_k§D0oPEjrvlC_%!*&L%gtmvyw§$9gf0Z&i#-2pl~b?!G&N&H!tDxyd4r§b\/zYjCTT1N!Ec6ZzFjIL47y3§Zux0vE+E2g!§E9=P1zyth0wMd%Dmzj75rXl\g`

<img width="411" src="https://github.com/user-attachments/assets/e2215b33-9772-472e-ac67-c2d3b87df3a3">

However, after rebooting the client, this password is no longer accepted when trying to start a remote session.
Also, when displaying the "Set permanent password" dialog after the reboot, no bullet characters are shown:

<img width="413" src="https://github.com/user-attachments/assets/da28c8d9-65d4-4e2f-998d-e83f47e18908">

### How to Reproduce

1. Set client's Permanent Password to
`D5K2Ulg_k§D0oPEjrvlC_%!*&L%gtmvyw§$9gf0Z&i#-2pl~b?!G&N&H!tDxyd4r§b\/zYjCTT1N!Ec6ZzFjIL47y3§Zux0vE+E2g!§E9=P1zyth0wMd%Dmzj75rXl\g`
2. Reboot client
3. Try starting a session with the client using the above password

### Expected Behavior

Check for passwords not being saved or altered due to unintended length or character limitations.
Remove unintended limitations or communicate limitations to users through password constraints/checks.

### Operating system(s) on local side and remote side

Windows 11 -> Windows 11

### RustDesk Version(s) on local side and remote side

1.3.0 -> 1.3.0

### Screenshots

Included in Bug Description section.

### Additional Context

Host:
OS: Windows 11 Pro
OS Build: 22631.4112
CPU: Intel Core i5-13600KF

Client:
OS: Windows 11 Pro
OS Build: 22631.4037
CPU: Intel Core i5-8259U

**Comment by felix-redlich** [comment]
This entry was moved from Issues to Discussions immediately after its creation. I do not understand why.
The issue prevents the software from functioning as intended without giving users an explanation for this behavior.

In Discussions, I then found #8666 and #9005 describing the same issue, with #8666 also having been downgraded from Issue to Discussion without explanation.


**"password" field in RustDesk.toml**

In #9005, the repository owner(s)(?) commented recommending to check:
`C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config\RustDesk.toml`

When setting the Permanent Password to
`Ab1234-5678`
or a similar one the "password" field in this file gets updated with a hash of the password:

![image](https://github.com/user-attachments/assets/b44ccf38-fdf2-45d6-82c2-5e07c42ea6d1)

When entering
`D5K2Ulg_k§D0oPEjrvlC_%!*&L%gtmvyw§$9gf0Z&i#-2pl~b?!G&N&H!tDxyd4r§b\/zYjCTT1N!Ec6ZzFjIL47y3§Zux0vE+E2g!§E9=P1zyth0wMd%Dmzj75rXl\g`
in the UI which accepts this password without error, the "password" field in RustDesk.toml gets set to an empty string:

![image](https://github.com/user-attachments/assets/56900629-b1ea-4918-86da-2514af6a183f)

To me, this is clearly a bug because some passwords entered by the user are accepted by the UI while not actually being saved. This is an issue preventing proper usage of the software, if only after a reboot.

**Reply by rustdesk** [reply]
@21pages try out `D5K2Ulg_k§D0oPEjrvlC_%!*&L%gtmvyw§$9gf0Z&i#-2pl~b?!G&N&H!tDxyd4r§b\/zYjCTT1N!Ec6ZzFjIL47y3§Zux0vE+E2g!§E9=P1zyth0wMd%Dmzj75rXl\g`

**Reply by rustdesk** [reply]
@felix-redlich Apologize for downgrading, this is my fault since you have given very good and detailed evidence to reproduce. Our dev will handle this with highest priority.

**Reply by 21pages** [reply]
The length of the string `D5K2Ulg_k§D0oPEjrvlC_%!*&L%gtmvyw§$9gf0Z&i#-2pl~b?!G&N&H!tDxyd4r§b\/zYjCTT1N!Ec6ZzFjIL47y3§Zux0vE+E2g!§E9=P1zyth0wMd%Dmzj75rXl\g` is 128 characters. However, since some characters (like `§`) are non-ASCII, the total byte length is 133, exceeding the maximum allowed length of 128. If you use a shorter string, such as `D5K2Ulg_k§D0oPEjrvlC_%!*&L%gtmvyw§$9gf0Z&i#-2pl~b?!G&N&H!tDxyd4r§b\/zYjCTT1N!Ec6ZzFjIL47y3§Zux0vE+E2g!`, it will fit within the limit.

`

**Reply by rustdesk** [reply]
Report error on GUI.

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/pull/9248
